kustomize/README.md

```diff
@@ -0,0 +1,114 @@
+```
+
+# Setup and Start a new  OpenShift Local
+
+You can configure the amount of cpu cores, memory in MiB and disk size in GiB allocated to your OpenShift Local environment depending on the resources of your machine
+
+```bash
+crc setup
+crc config --help
+crc config set cpus 8
+crc config set memory 12000
+crc config set disk-size 200
+crc start
+```
+
+If you need to upgrade your OpenShift Local because of some version error, delete the cluster and create it again like above. 
+
+```bash
+crc delete
+```
+
+# Setting up the OpenShift Environment for FIWARE using Kustomize GitOps
+
+## Login to OpenShift
+
+- Visit your OpenShift environment at the URL provided by the `crc start` command: https://console-openshift-console.apps-crc.testing
+- Login with username "kubeadmin", and the password provided in the `crc start` command. 
+- Login to the cli for your OpenShift Local cluster by clicking on "kubeadmin" in the top right corner and selecting "Copy login command". 
+- Click "Display token"
+- Copy the part that starts with `oc login --token=... --server=...` to the clipboard and paste it into the terminal and press [ Enter ] to login to the CLI. 
+
+## Clone the fiware-marinera repo
+
+```bash
+git clone git@github.com:rh-impact/fiware-marinera.git ~/.local/src/fiware-marinera
+cd ~/.local/src/fiware-marinera
+oc apply -k kustomize/overlays/openshift-local/
+```
+
+## Setup a new vault
+
+- Visit the vault here: https://vault-ui-vault.apps-crc.testing
+- Setup a new vault and keep track of the Initial Root Token, and the vault keys (1 key is enough). 
+- Click [ Enable new engine + ]
+- Select Generic -> KV, then click [ Next ]
+- Path: fiware
+- Click [ Enable Engine ]
+
+## Setup a Vault read policy
+
+- In the Vault, click Policies
+- Click [ Create ACL policy + ]
+- Name: vault-secret-reader
+- Here is the policy: 
+
+```
+path "/fiware/data/*" {
+  capabilities = ["read"]
+}
+```
+
+## Setup a new Access Authentication Method in vault
+
+- In the Vault, click Access
+- In Auth Methods, click [ Enable new method + ]
+- In "Infra", click "Kubernetes"
+- Path: kubernetes
+- Click [ Enable Method ]
+
+### Setup a secret-reader Auth Role
+
+- In the new `kubernetes/openshift-local` Authentication Method, click [ Create role + ]
+- Name: secret-reader
+- Bound service account names: vault-secret-reader
+- Bound service account namespaces: external-secrets-operator
+- Click "Tokens"
+- Generated token's policies: vault-secret-reader
+- Click [ Save ]
+
+## Setup new mongodb-secret in vault
+
+- In Vault, click Secrets
+- In the "fiware" path, add a new secret with a path of "mongodb-secret"
+- Add a password for the key: mongodb-password
+- Add a password for the key: mongodb-replica-set-key
+- Add a password for the key: mongodb-root-password
+- Click [ Save ]
+
+## Setup new argocd-auth-secret in vault
+
+- In Vault, click Secrets
+- In the "fiware" path, add a new secret with a path of "argocd-auth-secret"
```

Comment:
I'm still working on it, but why  do you connect the local argo instance to you're (private?) keycloak instance? Wouldn't it be easier to simply delegate the auth for argo to openshift?

Reply:
I am open to suggestions here for sure. I was hoping to build a consistent authentication for all the components (already deployed Red Hat SSO) for ArgoCD, OpenShift, Vault and the FIWARE applications. I figured logging into ArgoCD with the kubeadmin wasn't ideal or something like that. But maybe I'm just not familiar yet on how to make OpenShift Auth the best choice. Feel free to change. With Kustomize you can always apply patches per overlay and extend to do it your own way custom. Maybe OpenShift by default can use the Fiware provided Keycloak instance for authentication everywhere. What do you think?

Reply:
@computate Hi, that sounds like a good idea. I take a look at that. Did you see my PR to your repo? I'm not able to add you as reviewer.

Reply:
@wistefan Thanks for your commits to the PR from the FIWARE perspective. I have merged them into this branch!